tests/test_ftpfs.py

```diff
@@ -165,16 +165,23 @@ def tearDown(self):
         os.mkdir(self._temp_path)
         super(TestFTPFS, self).tearDown()
 
+    @property
+    def _get_url_start(self):
+        if self.user == "anonymous":
+            return "ftp://127.0.0.1"
+        else:
+            return "ftp://{}:{}@127.0.0.1".format(self.user, self.pasw)
+
     def test_ftp_url(self):
-        self.assertTrue(self.fs.ftp_url.startswith("ftp://127.0.0.1"))
+        self.assertTrue(self.fs.ftp_url.startswith(self._get_url_start))
         
     def test_geturl(self):
         self.fs.makedir("foo")
         self.fs.create("bar")
         self.fs.create("foo/bar")
-        self.assertTrue(self.fs.geturl('foo') == "ftp://127.0.0.1:{}/foo".format(self.server.port))
-        self.assertTrue(self.fs.geturl('bar') == "ftp://127.0.0.1:{}/bar".format(self.server.port))
-        self.assertTrue(self.fs.geturl('foo/bar') == "ftp://127.0.0.1:{}/foo/bar".format(self.server.port))
+        self.assertTrue(self.fs.geturl('foo') == "{}:{}/foo".format(self._get_url_start, self.server.port))
```

Comment:
Is this `_get_url_start` necessary? All the other tests seem to use `"ftp://user:1234@{}:{}/foo".format(self.server.host, self.server.port)` ?

Reply:
It's sole purpose is to distinguish between anonymous and real user, as we get `ftp://host.com/path/file` instead of `ftp://anonymous@host.com/path/file`
Maybe it's not very clean, though. If so, I would like to see more elegant solution.

Reply:
Sorry, I should have been clearer - do any of the tests here actually use the anonymous user?

Reply:
Thanks for clarifying. That's my fault: I thought the tests do cover anonymous users.
Will be back with anonymous tests!